### PR TITLE
my first PR!

### PR DESCRIPTION
**Pull request by browe83**
@rewinfrey
Hey man, Here's me taking a swing at my first PR.  Thanks for your help.

### CODE REVIEW

File: src/App.js [hunk 1]
```diff
@@ -74,6 +74,7 @@ function App() {
     const isEquals = input.match(/=/);
     const isDecimal = input.match(/./);
     const isClear = input.match(/AC/);
+    const isFloat = input.match(/^\d*\.?\d*$/);
```
Comment (rewinfrey) [inline src/App.js]: This regex seems to allow matching against single integer values, e.g. `3`. Is that intended? The name `isFloat` makes me think this regex would only allow matching on floating point numbers, but not simple integers.

**Review by rewinfrey — COMMENTED** [review]
This is looking good 👍 , I left a question for you about the intended matching for the `isFloat` regex.

**Comment by browe83** [comment]
@rewinfrey Hey man, hope you had a nice weekend.  My bad on not updating the calendar.  I think I have this project done but still need to clean up some of the conole.logs and unused variables.  To speak to your  question about `isFloat`, I ended up not using it.  Anyway, if you have sec to give it a look over, let me know what the next typical workflow step would be with the pull request.  For example, would I merge it into the main break after you approve and I clean up the code? Thanks!

**Comment by rewinfrey** [comment]
Hey @browe83 thanks for the ping! Usually a PR is used to signal to your teammates or others that you have a small-ish code change and you'd like for someone else to review it. PR review is a broad topic, but usually it's good to keep in mind:
* Does the code do what it says on the tin (usually based on the PR description)?
* Does the code match / conform to the code standard in the project?
* Is there confidence the code is correct (in terms of behavior and business logic)? Tests are great for verifying behavior, business logic usually requires some understanding about the requirements / domain knowledge.

So with that in mind, I'll take a look at this PR and review it according to ☝️ later this evening. Because PR comments are in text form, and nuance is lost in text, I try to be extra careful with the wording of feedback. I like to frame everything in the form of a question -- and clarify before suggesting a change. I try to pay particular attention to language that can be disheartening, words like "just", "simply", or very terse comments / questions can often be misunderstood or come across as judgement. When receiving review it's always helpful to remember not to assume intent, and when giving review it's always good to check for words that can imply judgement. Also emoji's go a long way towards helping set a "tone" for the review 😄

File: src/App.js [hunk 2]
```diff
@@ -63,73 +63,187 @@ const useStyles = makeStyles({
...
-      return;
+  const endsWithOperand = /\d$/;
+  const endsWithOperator = /[+-/*]$/;
+  const isDecimal = /^\d*\.?\d*$/;
+  const endsWithNegative = /-$/;
+  const isNegative = /-/;
+  const endsWithNonNegativeOperator = /[*+/]$/;
+  const isOperator = /[*/+-]/;
+  const isMultiply = /X/;
+  const isOperand = /\d/;
```
Comment (rewinfrey) [inline src/App.js]: I wonder about the `isOperand` name because it doesn't include floating point numbers, and floating point numbers could be operands in this calculator. What do you think about a similarly named `isInteger` here to go along with `isDecimal`?

File: src/App.js [hunk 3]
```diff
@@ -63,73 +63,187 @@ const useStyles = makeStyles({
...
+      console.log('condition 5 operator');
+      setCurrentToken(input);
+      setEquation(equation.slice(0, -1) + input);
+    } else if (equation === '') {
+      setCurrentToken(input);
+      setEquation(input);
+    } else if (equation.includes('=')) {
+      console.log('handleOperator condition 6');
+      setCurrentToken(input);
+      setEquation(answer + input);
```
Comment (rewinfrey) [inline src/App.js:143]: With the long `else if` chain it is hard to read at a glance. What do you think about defining a function for each of the test conditions with a descriptive name? The idea would be something that reads like:

```javascript
// Example of helper function for use in test conditions in `handleOperator`:
const function addOperator() { 
  return endsWithOperand.test(equation) && !equation.includes('=')
}

...

// Inside the handleOperator function:
if (addOperator()) {
  setCurrentToken(input);
  setEquation(equation + input);
  // return early
  return 
}
.
.
.
// Each test condition can be its own if statement.
if (...) {
...
}
// end of handleOperator function
```

File: src/App.js [hunk 4]
```diff
@@ -63,73 +63,187 @@ const useStyles = makeStyles({
...
+
+  const handleEval = ({ currentTarget: { value } }) => {
+    const input = value;
+    console.log('eval:', input);
+    // const numEqualsNum = /\d+=\d+/;
+    const startsWithMultiOrDiv = /^[*/]/;
+    if (!equation.includes('=') && !startsWithMultiOrDiv.test(equation)) {
+      console.log('handleEval condition 1');
+      // eslint-disable-next-line no-eval
+      const result = eval(equation);
```
Comment (rewinfrey) [inline src/App.js:200]: I think this is fine for this project, but `eval` in any language that evaluates an arbitrary string of executable code is considered highly dangerous to use. You may have seen [Never use eval!](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/eval#never_use_eval!) before. If you want an extra challenge, it's a fun challenge to do the evaluation of the equation input string in place of using `eval`.

**Review by rewinfrey — APPROVED** [review]
This looks like a functioning calculator app! 🎉 👍  Nice job. I left you some feedback to consider, but wouldn't consider a blocker for this project.

If this were for a production system I might require there be tests for the JavaScript functions and regexes to ensure that edge cases have been exercised, but not necessary here. However, if you're itching to get into automated testing, this would possibly be a great project to write some validation tests using a JavaScript unit test library, e.g. [jest](https://www.testim.io/blog/jest-testing-a-helpful-introductory-tutorial/).
